cmd/bpf2go/gen/output.go

```diff
@@ -161,30 +161,40 @@ func Generate(args GenerateArgs) error {
 		Identifier: args.Identifier,
 	}
 
+	var typeDecls []string
+	needsStructsPkg := false
+	for _, typ := range types {
+		name := typeNames[typ]
+		decl, err := gf.TypeDeclaration(name, typ)
+		if err != nil {
+			return fmt.Errorf("generating %s: %w", name, err)
+		}
+		needsStructsPkg = needsStructsPkg || strings.Contains(decl, " _ structs.HostLayout; ")
```

Comment:
This can't quietly cause a regression, but I'd prefer a type assertion on `*btf.Struct` instead. Matching on a string including whitespace looks a little fragile in any case.

Reply:
Great idea, thanks!